src/System.Xml.XDocument/src/System/Xml/Linq/XLinq.cs

```diff
@@ -559,16 +559,8 @@ internal void WriteStreamingElement(XStreamingElement e)
             FlushElement();
             _element = e;
             Write(e.content);
-            bool contentWritten = _element == null;
```

Comment:
cc: @sepidehMS 
We just dereferenced e after assigned _element to e, so we know _element isn't null